native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.android.js

```diff
@@ -168,108 +139,63 @@ const BpkBannerAlert = props => {
   } = props;
 
   const expandable = children !== null;
-  const { iconSource, iconStyle } = ALERT_TYPE_STYLES[type] || {};
+  const { icon, iconStyle } = ALERT_TYPE_STYLES[type] || {};
 
-  const outerStyle = [styles.flexRow];
-  if (bannerStyle) {
-    outerStyle.push(bannerStyle);
-  }
-
-  const mainContent = (
-    <View style={outerStyle}>
-      <View style={styles.mainContainer}>
-        <View style={styles.flexRow}>
-          <BpkIcon style={[styles.icon, iconStyle]} icon={iconSource} small />
-          <BpkText style={styles.text} textStyle="sm">
-            {message}
-          </BpkText>
-        </View>
-        {expandable && (
-          <ExpandableContent expanded={expanded}>{children}</ExpandableContent>
-        )}
-      </View>
-      {dismissable && (
-        <DismissableButton
-          dismissButtonLabel={dismissButtonLabel}
-          onDismiss={onDismiss}
-        />
-      )}
-      {expandable && <ExpandableIcon expanded={expanded} />}
-    </View>
+  const ExpandToggleView = expandToggleViewProps => (
```

Comment:
Can we move this outside? Recreating the function for each render seems redundant and the value of capturing a few values in the closure seems negligable at best

Reply:
Fixed!